### ISSUE TITLE
Make threadpool queue size configurable

### ISSUE THREAD

**Issue by erip** [bab2min/tomotopy#210]
Analysis [here](https://github.com/bab2min/tomotopy/issues/182#issuecomment-1715862530) suggests that the queue size is too small to be effective for parallel processing. It's a bit hard to follow the code path, but it seems to be [capped at `8*num_threads`](https://github.com/bab2min/tomotopy/blob/6ef712c63bad504801a981a4a52aa938c80f0bda/src/TopicModel/LDAModel.hpp#L917) in the case of LDA. While it isn't obvious what a good default is, this being hard coded seems to be causing issues. It should be a kwarg from `infer` so users can tune inference.

**Comment by erip** [comment]
I guess for what it's worth: I wanted to submit a PR for this change, but can't quite get the code to build locally on my WSL rig...

**Comment by jgb-hda** [comment]
For your information: I did manage to compile tomotopy with your proposed changes, however this did not improve the slow inference. 

I don't have precise timings or any nice graphs. but it felt as it still took as long as before. I did change 8 to 128. No idea if that is actually a good value, but it didn't seem to change anything anyways, as the CPU was still under high load but not doing much, as it didn't got warm, so I guess it still did spend most of its time waiting …
